core/base/metricDistortion/MetricDistortion.h

```diff
@@ -0,0 +1,198 @@
+                            std::vector<std::vector<double>> &distanceMatrix,
+                            std::vector<double> &surfaceArea,
+                            std::vector<double> &metricArea,
+                            std::vector<double> &ratioArea) {
+      unsigned int dim = surfaceCells.size();
+      surfaceArea = std::vector<double>(dim, std::nan(""));
+      metricArea = std::vector<double>(dim, std::nan(""));
+      ratioArea = std::vector<double>(dim, std::nan(""));
+      for(unsigned int i = 0; i < dim; ++i) {
+        if(surfaceCells[i].size() < 3 or surfaceCells[i].size() > 4)
+          continue;
+
+        int i0 = surfaceCells[i][0];
+        int i1 = surfaceCells[i][1];
+        int i2 = surfaceCells[i][2];
+
+        surfaceArea[i] = triangleArea3D(
+          surfacePoints[i0], surfacePoints[i1], surfacePoints[i2]);
+
+        if(distanceMatrix.size() != 0)
+          metricArea[i] = triangleAreaFromSides(distanceMatrix[i0][i1],
+                                                distanceMatrix[i1][i2],
+                                                distanceMatrix[i2][i0]);
+
+        if(surfaceCells[i].size() == 4) {
+          int i3 = surfaceCells[i][3];
+          surfaceArea[i] += triangleArea3D(
+            surfacePoints[i1], surfacePoints[i2], surfacePoints[i3]);
+          if(distanceMatrix.size() != 0)
+            metricArea[i] += triangleAreaFromSides(distanceMatrix[i1][i2],
+                                                   distanceMatrix[i2][i3],
+                                                   distanceMatrix[i3][i1]);
+        }
+
+        if(distanceMatrix.size() != 0)
+          ratioArea[i] = metricArea[i] / surfaceArea[i];
+      }
+    }
+
+    //-------------------------------------------------------------------------
+    // Utils
+    //-------------------------------------------------------------------------
+    double cosineLaw(double a, double b, double c) {
+      // Get the angle opposite to edge c
+      return std::acos((a * a + b * b - c * c) / (2.0 * a * b));
+    }
+
+    double l2Distance(std::vector<double> &p1, std::vector<double> &p2) {
+      double distance = 0.0;
+      for(unsigned int i = 0; i < p1.size(); ++i)
+        distance += std::pow(p1[i] - p2[i], 2);
+      return std::sqrt(distance);
+    }
+
+    double triangleArea3D(std::vector<double> &x1,
+                          std::vector<double> &x2,
+                          std::vector<double> &x3) {
+      std::vector<double> AB(3), AC(3);
+      for(unsigned int i = 0; i < 3; ++i) {
+        AB[i] = x2[i] - x1[i];
+        AC[i] = x3[i] - x1[i];
+      }
+      return std::sqrt(std::pow(AB[1] * AC[2] - AB[2] * AC[1], 2)
+                       + std::pow(AB[2] * AC[0] - AB[0] * AC[2], 2)
+                       + std::pow(AB[0] * AC[1] - AB[1] * AC[0], 2))
+             / 2.0;
+    }
+
+    double triangleAreaFromSides(double a, double b, double c) {
```

Comment:
Please move this to `core/base/geometry/Geometry.h`

Reply:
Done in the next commit!